src/plugins/kibana_utils/common/persistable_state/migrate_to_latest.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { compare } from 'semver';
+import { SerializableState, VersionedState, MigrateFunctionsObject } from './types';
+
+export function migrateToLatest<S extends SerializableState>(
+  migrations: MigrateFunctionsObject,
+  [state, oldVersion]: VersionedState
+): VersionedState<S> {
+  const versions = Object.keys(migrations || {})
+    .sort(compare)
+    .filter((v) => compare(v, oldVersion) > 0);
+
+  if (!versions.length) return [state, oldVersion] as VersionedState<S>;
+
+  for (const version of versions) {
+    state = migrations[version]!(state);
+  }
+
+  return [state as S, versions[versions.length - 1]];
```

Comment:
you should return current version here as well

Reply:
Not sure what you mean, I think it should return the latest version to which it was migrated, which is `versions[versions.length - 1]`.

Reply:
we are always migrating to latest version. the fact that migrations were not registered for some versions is just not important here.

for example i want to migrate embeddable 7.12.0 to latest .... latest is 7.14.0 ... i don't want my embeddable to stay at version 7.13.2 just because there is no migration necesart for 7.14.0, we can assume version to be 7.14.0.

Reply:
I see what you mean now: always return some global *current version* regardless to what actual version the object was migrated to.

But I don't think we should do that. Or if we do that, it should not be part of this function. I guess we can discuss that.